test/controllers/workflow_executions_controller_test.rb

```diff
@@ -58,4 +58,54 @@ class WorfklowExecutionsControllerTest < ActionDispatch::IntegrationTest
   end
 
   # rubocop:enable Naming/VariableNumber
+
+  test 'should not delete a prepared workflow' do
+    workflow_execution = workflow_executions(:irida_next_example_prepared)
+    assert workflow_execution.prepared?
+    assert_difference -> { WorkflowExecution.count } => 0,
+                      -> { SamplesWorkflowExecution.count } => 0 do
+      delete workflow_execution_path(workflow_execution, format: :turbo_stream)
+    end
+    assert_response :unprocessable_entity
+  end
+
+  test 'should delete a completed workflow' do
+    workflow_execution = workflow_executions(:irida_next_example_completed)
+    assert workflow_execution.completed?
+    assert_difference -> { WorkflowExecution.count } => -1,
+                      -> { SamplesWorkflowExecution.count } => -1 do
+      delete workflow_execution_path(workflow_execution, format: :turbo_stream)
+    end
+    assert_response :success
+  end
+
+  test 'should delete an errored workflow' do
+    workflow_execution = workflow_executions(:irida_next_example_error)
+    assert workflow_execution.error?
+    assert_difference -> { WorkflowExecution.count } => -1,
+                      -> { SamplesWorkflowExecution.count } => -1 do
+      delete workflow_execution_path(workflow_execution, format: :turbo_stream)
+    end
+    assert_response :success
+  end
+
+  test 'should not delete a canceling workflow' do
+    workflow_execution = workflow_executions(:irida_next_example_canceling)
+    assert workflow_execution.canceling?
+    assert_difference -> { WorkflowExecution.count } => 0,
+                      -> { SamplesWorkflowExecution.count } => 0 do
+      delete workflow_execution_path(workflow_execution, format: :turbo_stream)
+    end
+    assert_response :unprocessable_entity
+  end
+
+  test 'should delete a canceled workflow' do
+    workflow_execution = workflow_executions(:irida_next_example_canceled)
+    assert workflow_execution.canceled?
+    assert_difference -> { WorkflowExecution.count } => -1,
+                      -> { SamplesWorkflowExecution.count } => -1 do
+      delete workflow_execution_path(workflow_execution, format: :turbo_stream)
+    end
+    assert_response :success
+  end
```

Comment:
Can I get you to add tests for `submitting`, `new`, `queued`, and `running` as well

Reply:
2e4c95fa7d54e38f495de3a628c534eb7d1a7a42

Reply:
I tried creating my own state :laughing: Thanks!